### PR TITLE
(test) Fix test_agent_controller.py mock exceptions

### PR DESCRIPTION
**Pull request by tobitege**
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?**

The `test_agent_controller.py` unit test generated several async-related errors due to missing `close()` statements for the `controller` object.

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

- Added `controller.close()` to tests.

---
**Other references**

These unit tests were introduce with #3357

### CODE REVIEW

**Review by neubig — CHANGES_REQUESTED** [review]
Thank you for working with this! I'm all in favor of fixing errors, but do you have any examples of these errors actually happening? It seems that CI has been passing, and on my local machine I also haven't had any issues running these tests.

My "request changes" is because I'd like to avoid adding any code specifically to satisfy tests into the core library, and if it was possible for me to reproduce the error I could help suggest an alternative.

**Comment by tobitege** [comment]
Without the "close()" this would appear in the unit test:
```Task was destroyed but it is pending!
task: <Task pending name='Task-16' coro=<AgentController._start_step_loop() running at /mnt/d/github/OpenDevin/openhands/controller/agent_controller.py:154> wait_for=<Future pending cb=[Task.task_wakeup()]>>```

File: openhands/controller/agent_controller.py [hunk 1]
```diff
@@ -130,7 +130,8 @@ def update_state_before_step(self):
 
     async def update_state_after_step(self):
         # update metrics especially for cost
-        self.state.local_metrics = self.agent.llm.metrics
+        if hasattr(self.agent, 'llm'):  # for unit test
+            self.state.local_metrics = self.agent.llm.metrics
```
[the remark this answers is not in the document]
Comment (tobitege) [reply openhands/controller/agent_controller.py:134]: Mock exception looks like this for test `test_on_event_message_action`:
```py
Traceback (most recent call last):
  File "/mnt/d/github/OpenDevin/openhands/controller/agent_controller.py", line 154, in _start_step_loop
    await self._step()
  File "/mnt/d/github/OpenDevin/openhands/controller/agent_controller.py", line 488, in _step
    await self.update_state_after_step()
  File "/mnt/d/github/OpenDevin/openhands/controller/agent_controller.py", line 134, in update_state_after_step
    self.state.local_metrics = self.agent.llm.metrics
                               ^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/unittest/mock.py", line 653, in __getattr__
    raise AttributeError("Mock object has no attribute %r" % name)
AttributeError: Mock object has no attribute 'llm'
23:08:41 - openhands:ERROR: agent_controller.py:160 - Error while running the agent: Mock object has no attribute 'llm'
23:08:41 - openhands:ERROR: agent_controller.py:161 - Traceback (most recent call last):
  File "/mnt/d/github/OpenDevin/openhands/controller/agent_controller.py", line 154, in _start_step_loop
    await self._step()
  File "/mnt/d/github/OpenDevin/openhands/controller/agent_controller.py", line 488, in _step
    await self.update_state_after_step()
  File "/mnt/d/github/OpenDevin/openhands/controller/agent_controller.py", line 134, in update_state_after_step
    self.state.local_metrics = self.agent.llm.metrics
                               ^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/unittest/mock.py", line 653, in __getattr__
    raise AttributeError("Mock object has no attribute %r" % name)
AttributeError: Mock object has no attribute 'llm'
```
Will look into mocking these `llm.metrics` attributes.

**Comment by tobitege** [comment]
With the fix in the test script itself, the change in the `agent_controller.py` is no longer needed, so reverted that.
This should be good now.

**Review by neubig — APPROVED** [review]
Thanks!

**Pull request merged**